tests/unit/s2n_tls13_server_cert_selection_test.c

```diff
@@ -0,0 +1,226 @@
+/*
```

Comment:
I think it's okay for this unit test to not be exhaustive since I think we can get more coverage(with less dev time) in our integ test that iterates over all combinations of {client_sigs, server_certs}

Reply:
It might also be nice to use "hybrid" certs in the tests where the key type and cert signature type don't match.